pkg/dynamicinstrumentation/diconfig/binary_inspection.go

```diff
@@ -69,6 +69,11 @@ func AnalyzeBinary(procInfo *ditypes.ProcessInfo) error {
 		}
 	}
 
+	// do not run InspectWithDWARF if there are no fieldIDs to inspect
+	if len(fieldIDs) == 0 {
+		return nil
+	}
+
 	r, err := bininspect.InspectWithDWARF(elfFile, functions, fieldIDs)
```

Comment:
This check is unnecessary. `InspectWithDWARF` takes fieldIDs as an option because while it's inspecting DWARF info for `functions`, it's more efficient to also inspect for struct fields. But it's possible/likely fieldIDs is empty if the parameters of the functions being inspected aren't structs.

Reply:
Makes sense, I've removed the commit.

FWIW this change was in the code when we ran the analysis tests so it will have affected the results there.